ecleankernel/file.py

```diff
@@ -91,6 +95,60 @@ def __init__(self,
         super().__init__(path, KernelFileType.KERNEL)
         self.internal_version = self.read_internal_version()
 
+    def __decompress_raw(self) -> bytes:
+        f = open(self.path, 'rb')
+        magic = [
```

Comment:
Please use a dict and don't rely on hardcoding indices. E.g.:

```
magic = {
    b'...': 'gzip',
    ...
```

Then use a `for` loop to determine which one matched, and go forward from these strings.

Or you could go a bit further and use strings matching module names, and then for most of the options you'd use something like:

```
mod = importlib.import_module(name)
b = getattr(mod, 'decompress')(f.read())
```

and you'd probably only need an exception for zstd.

Reply:
That's exactly what I wanted to do, but I didn't know you could dynamically import and call functions by variable - I still have a lot of python to learn, thanks!